### PR TITLE
Add migration postgresql script from 10 to 13

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
## What does this PR change?

Add migration postgresql script from 10 to 13

## GUI diff

No difference.

- [X] **DONE**

## Documentation

- Documentation issue was created: https://github.com/uyuni-project/uyuni-docs/issues/968 https://github.com/SUSE/spacewalk/issues/14959
https://github.com/uyuni-project/uyuni-docs/pull/969

- [x] **DONE**

## Test coverage
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added
- Cucumber tests were added

- [ ] **DONE**

## Links

Fixes https://github.com/SUSE/spacewalk/issues/14926

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [x] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Comment by jordimassaguerpla** [comment]
@Bischoff What should I  do with the testing part? Should this be added to the manual tests? Is there any cucumber for testing the migration from 10 to 12 that I could adapt for 10 to 13?

**Comment by jordimassaguerpla** [comment]
How to test it:

Follow this https://documentation.suse.com/external-tree/en-us/suma/4.2/suse-manager/upgrade/server-y-zypper.html by replacing 1 and 2 by doing this:

1- Add SP3 and manager 4.2 repos (from download.suse.de)
2- disable sp1 and 4.0
3- run zypper dup

File: susemanager/bin/pg-migrate-10-to-13.sh [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    POSTGRES_LANG="en_US.UTF-8"
+fi
+su -s /bin/bash - postgres -c "initdb -D /var/lib/pgsql/data --locale=$POSTGRES_LANG"
+if [ $? -eq 0 ]; then
+    echo "`date +"%H:%M:%S"`   Successfully initialized new postgresql $NEW_VERSION database."
+else
+    echo "`date +"%H:%M:%S"`   Initialization of new postgresql $NEW_VERSION database failed!"
+    echo "`date +"%H:%M:%S"`   Trying to restore previous state..."
+    mv /var/lib/pgsql/data /var/lib/pgsql/data-new-failed
+    mv /var/lib/pgsql/data-pg10 /var/lib/pgsql/data
```
Comment (juliogonzalez) [inline susemanager/bin/pg-migrate-10-to-13.sh:95]: What about using `${OLD_VERSION}` here as well?

File: susemanager/bin/pg-migrate-10-to-13.sh [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    /usr/sbin/update-alternatives --set postgresql /usr/lib/postgresql10
```
Comment (juliogonzalez) [inline susemanager/bin/pg-migrate-10-to-13.sh:96]: And here.

File: susemanager/bin/pg-migrate-10-to-13.sh [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+echo "`date +"%H:%M:%S"`   Upgrade database to new version postgresql $NEW_VERSION..."
+su -s /bin/bash - postgres -c "pg_upgrade --old-bindir=/usr/lib/postgresql$OLD_VERSION/bin --new-bindir=/usr/lib/postgresql$NEW_VERSION/bin --old-datadir=/var/lib/pgsql/data-pg$OLD_VERSION --new-datadir=/var/lib/pgsql/data $FAST_UPGRADE"
+if [ $? -eq 0 ]; then
+    echo "`date +"%H:%M:%S"`   Successfully upgraded database to postgresql $NEW_VERSION."
+else
+    echo "`date +"%H:%M:%S"`   Upgrading database to version $NEW_VERSION failed!"
+    echo "`date +"%H:%M:%S"`   Trying to restore previous state..."
+    mv /var/lib/pgsql/data /var/lib/pgsql/data-new-failed
+    mv /var/lib/pgsql/data-pg$OLD_VERSION /var/lib/pgsql/data
```
Comment (juliogonzalez) [inline susemanager/bin/pg-migrate-10-to-13.sh:108]: I strongly recommend you always use` ${NAME}. It improves readability, and also is safer as it works regardless of what's around the variable :-)

**Review by juliogonzalez — COMMENTED** [review]
Some comments.

File: susemanager/bin/pg-migrate-10-to-13.sh (hunk 1 above)
Comment (jordimassaguerpla) [reply susemanager/bin/pg-migrate-10-to-13.sh]: fixed

File: susemanager/bin/pg-migrate-10-to-13.sh (hunk 2 above)
Comment (jordimassaguerpla) [reply susemanager/bin/pg-migrate-10-to-13.sh]: fixed

File: susemanager/bin/pg-migrate-10-to-13.sh (hunk 3 above)
Comment (jordimassaguerpla) [reply susemanager/bin/pg-migrate-10-to-13.sh:108]: yeah, I thought about that, but then I have to change that to the rest of the script to be consistent, and I thought it was too much refactor at this moment

**Comment by jordimassaguerpla** [comment]
> How to test it:
> 
> Follow this https://documentation.suse.com/external-tree/en-us/suma/4.2/suse-manager/upgrade/server-y-zypper.html by replacing 1 and 2 by doing this:
> 
> 1- Add SP3 and manager 4.2 repos (from download.suse.de)
> 2- disable sp1 and 4.0
> 3- run zypper dup

Tests pass.

Seems the migration path is enabled in SCC, so I will test that as well.

**Comment by jordimassaguerpla** [comment]
With SCC tests passed as well.
So we can say this works.

**Comment by jordimassaguerpla** [comment]
> @Bischoff What should I do with the testing part? Should this be added to the manual tests? Is there any cucumber for testing the migration from 10 to 12 that I could adapt for 10 to 13?

I added a test to your manual tests: https://gitlab.suse.de/galaxy/testplan/-/merge_requests/24, hope this is ok

**Review by juliogonzalez — APPROVED** [review]

**Pull request merged**